tmp36-temperature-sensor/include/TemperatureSensor.h

```diff
@@ -0,0 +1,17 @@
+#ifndef TMP36_TEMPERATURE_SENSOR_H
+#define TMP36_TEMPERATURE_SENSOR_H
+
+#include <AnalogPinReader.h>
+
+namespace wlp {
+    namespace TMP36 {
+        class TemperatureSensor: public AnalogPinReader {
+        public:
+            explicit TemperatureSensor(Board::AnalogPin analog_pin)
+            : AnalogPinReader(analog_pin) {};
+            virtual double read_value() override;
```

Comment:
does this need to be virtual?

Reply:
also realised some of the other functions for the other sensors are like this too

Reply:
You did ask me to make it virtual in #2 and #3

Reply:
I thought it was an error to `override` a `virtual` with non-`virtual`, but apparently it just makes both methods virtual.

Reply:
usually a function is marked `virtual` when it's meant to be overridden (more of a code readability issue) so unless you plan to override `read_value` for the sensors `virtual` should be removed

I'll merge this now and fix after